snuba/query/snql/parser.py

```diff
@@ -457,9 +457,9 @@ def visit_where_clause(
         return conditions
 
     def visit_having_clause(
-        self, node: Node, visited_children: Tuple[Any, Any, Expression]
+        self, node: Node, visited_children: Tuple[Any, Any, Any, Expression]
     ) -> Expression:
-        _, _, conditions = visited_children
+        _, _, _, conditions = visited_children
```

Comment:
Is this change actually related to the NOT IN / NOT LIKE change above ?

Reply:
The changes are linked incidentally. The discover test for the cached failure revealed that we weren't sending `having` clauses through conftest. Fixing that revealed that SnQL was breaking on the HAVING clause. Fixing _that_ revealed a different test that had a NOT IN in the HAVING clause, which we weren't handling correctly.

So while in theory the fixes are independent, I couldn't fix one without fixing the others.

Reply:
Yes, indirectly. This is fixing the having clause in general. The NOT IN change fixes a test that breaks once the having clause is fixed: `test_tags_key_boolean_condition` which has a NOT IN in the having clause.